api/runner/runner.go

```diff
@@ -29,6 +29,7 @@ type Runner struct {
 	flog         FuncLogger
 	availableMem int64
 	usedMem      int64
+	Tasks        chan task.Request
```

Comment:
if tasks are internal to package `runner`, why are you making it visible? Also consider that, once you make this channel something internal to package `runner`, you are also removing from test procedures to actually intercept the communication between `server` and `runner`. Is this what you want? If so, could you please elaborate what are the advantages in terms of testing of not having the possibility to intercept the communication? 

In the current test suite, we actually intercept the communication - it makes the tests faster, and prevents unnecessary interfacing with the testing docker, thus helping with separation of concerns during test execution.